### PR TITLE
Add a script to set up shell aliases for linutil

### LINKED ISSUE #349: Add alias for linutil in bashrc
Can someone add a simple alias to .bashrc

alias linutil="curl -fsSL https://christitus.com/linux | sh"

### PR DESCRIPTION
**Pull request by adamperkowski**
# Add a script to set up shell aliases for linutil

## Type of Change
- [x] New feature

## Description
ref #349, https://github.com/ChrisTitusTech/linutil/issues/349#issuecomment-2346816737

## Impact
Created a new menu entry and sh script.

## Issue related to PR
- Resolves #349

## Checklist
- [x] My code adheres to the coding and style guidelines of the project.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] My changes generate no errors/warnings/merge conflicts.

### CODE REVIEW

**Comment by ChrisTitusTech** [comment]
Sorry for the inconvenience. We had a massive restructure of the codebase to improve future development. Because of this can you update your PR to the new structure. Thank you for your assistance and contribution.

**Comment by adamperkowski** [comment]
@ChrisTitusTech Updated. Everything works.

File: tabs/applications-setup/shell-aliases.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
+#!/bin/sh
+
+. ../common-script.sh
+
+if command_exists linutil; then
+    echo "linutil already exists in your system."
+    exit 1
+fi
```
Comment (lj3954) [inline tabs/applications-setup/shell-aliases.sh:8]:
```suggestion
```
We can move this to a precondition, in order for the script to not be displayed whatsoever in the case that the command already exists.

File: tabs/applications-setup/tab_data.toml [hunk 2]
```diff
@@ -27,3 +27,7 @@ script = "rofi-setup.sh"
 [[data]]
 name = "ZSH Prompt"
 script = "zsh-setup.sh"
+
+[[data]]
+name = "Shell aliases for linutil"
+script = "shell-aliases.sh"
```
Comment (lj3954) [inline tabs/applications-setup/tab_data.toml:33]:
```suggestion
script = "shell-aliases.sh"

[[data.preconditions]]
matches = false
data = "command_exists"
values = [ "linutil" ]
```

**Review by lj3954 — CHANGES_REQUESTED** [review]

File: tabs/applications-setup/tab_data.toml (hunk 2 above)
Comment (adamperkowski) [reply tabs/applications-setup/tab_data.toml]: > We can move this to a precondition, in order for the script to not be displayed whatsoever in the case that the command already exists.

That's a good idea. Done.

File: tabs/applications-setup/shell-aliases.sh [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+shells=(bash zsh fish tcsh ksh nu)                                                    # Supported shells
+configs=(.bashrc .zshrc config.fish .tcshrc .kshrc config.nu)                         # Shell config filenames
+aliases=("alias linutil=\"curl -fsSL https://christitus.com/linux | sh\""             # Alias structures for different shell configs
+         "alias linutil=\"curl -fsSL https://christitus.com/linux | sh\""
+         "alias linutil \"curl -fsSL https://christitus.com/linux | sh\""
+         "alias linutil \"curl -fsSL https://christitus.com/linux | sh\""
+         "alias linutil='curl -fsSL https://christitus.com/linux | sh'"
+         "alias linutil = curl -fsSL https://christitus.com/linux | sh")
+
+for ((i=0; i<${#shells[@]}; i++)); do
```
Comment (nnyyxxxx) [inline tabs/applications-setup/shell-aliases.sh:14]: This is not posix-compliant.

File: tabs/applications-setup/shell-aliases.sh [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    if command_exists ${shells[$i]}; then
```
Comment (nnyyxxxx) [inline tabs/applications-setup/shell-aliases.sh:15]: This is not posix-compliant.

File: tabs/applications-setup/shell-aliases.sh [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+        echo "Found ${shells[$i]}. Adding the alias..."
```
Comment (nnyyxxxx) [inline tabs/applications-setup/shell-aliases.sh:16]: This is not posix compliant.

File: tabs/applications-setup/shell-aliases.sh [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+        config_file="$HOME/${configs[$i]}"
```
Comment (nnyyxxxx) [inline tabs/applications-setup/shell-aliases.sh:17]: This is not posix compliant.

File: tabs/applications-setup/shell-aliases.sh [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+        if [ ! -f "$config_file" ]; then
+            if [ "${shells[$i]}" == "fish" ] || [ "${shells[$i]}" == "nu" ]; then     # Change config dirs for specific shells
```
Comment (nnyyxxxx) [inline tabs/applications-setup/shell-aliases.sh:19]: This is not posix compliant.

File: tabs/applications-setup/shell-aliases.sh [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+                config_dir="$XDG_CONFIG_HOME/${shells[$i]}"
```
Comment (nnyyxxxx) [inline tabs/applications-setup/shell-aliases.sh:20]: This is not posix compliant.

File: tabs/applications-setup/shell-aliases.sh [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+                mkdir -p "$config_dir"
+                config_file="$config_dir/${configs[$i]}"
```
Comment (nnyyxxxx) [inline tabs/applications-setup/shell-aliases.sh:22]: This is not posix compliant.

File: tabs/applications-setup/shell-aliases.sh [hunk 10]
```diff
@@ -0,0 +1,37 @@
...
+            elif [ "${shells[$i]}" == "bash" ] && [ -f "/etc/skel/.bashrc" ]; then    # Default distro-specific config for bash
```
Comment (nnyyxxxx) [inline tabs/applications-setup/shell-aliases.sh:23]: This is not posix compliant.

File: tabs/applications-setup/shell-aliases.sh [hunk 11]
```diff
@@ -0,0 +1,37 @@
...
+                cp "/etc/skel/.bashrc" "$config_file"
+            else
+                touch "$config_file"
+            fi
+        fi
+        if ! grep -Fxq "${aliases[$i]}" "$config_file"; then                          # Check if alias already exists
```
Comment (nnyyxxxx) [inline tabs/applications-setup/shell-aliases.sh:29]: This is not posix compliant.

File: tabs/applications-setup/shell-aliases.sh [hunk 12]
```diff
@@ -0,0 +1,37 @@
...
+            echo "${aliases[$i]}" >> "$config_file"
```
Comment (nnyyxxxx) [inline tabs/applications-setup/shell-aliases.sh:30]: This is not posix compliant.

**Comment by nnyyxxxx** [comment]
@adamperkowski Too many bashisms in this PR; this will have to get refactored.

File: tabs/applications-setup/shell-aliases.sh [hunk 13]
```diff
@@ -0,0 +1,37 @@
+#!/bin/sh
```
Comment (nnyyxxxx) [inline tabs/applications-setup/shell-aliases.sh:1]:
```suggestion
#!/bin/sh -e
```

**Comment by adamperkowski** [comment]
@nnyyxxxx Done.

File: tabs/applications-setup/shell-aliases.sh (hunk 13 above)
Comment (adamperkowski) [reply tabs/applications-setup/shell-aliases.sh:1]: image

**Review by nnyyxxxx — APPROVED** [review]
Looks great.

**Review by lj3954 — APPROVED** [review]

**Pull request closed without merging**